File: backend/Modules/Appointments/Modules.Appointments.Features/GetPatientAppointments/GetPatientAppointmentsDto.cs
using Modules.Appointments.Domain.Enums;
using Modules.Professionals.PublicApi.Contracts;

namespace Modules.Appointments.Features.GetPatientAppointments;

public record GetPatientAppointmentsDto( 
    Guid Id,
    Guid PatientId,
    Guid ProfessionalId,
    string? Notes,
    DateTime StartDate,
    DateTime EndDate,
    AppointmentUrgency Urgency,
    AppointmentStatus Status,
    decimal Price,
    DateTime? OfferedAt,
    DateTime? ConfirmedAt,
    DateTime? CompletedAt,
    DateTime? CancelledAt,
    DateTime CreatedAt,
    DateTime UpdatedAt,
    ProfessionalDto Professional);

File: frontend/src/features/patient/hooks/GetPatientAppointments/GetPatientAppointmentsDto.ts
import type {
  AppointmentStatus,
  AppointmentUrgency,
  ProfessionalDto,
} from "@/features/professional";

export interface GetPatientAppointmentsDto {
  id: string;
  patientId: string;
  professionalId: string;
  notes?: string;
  startDate: string;
  endDate: string;
  urgency: AppointmentUrgency;
  status: AppointmentStatus;
  price: number;
  offeredAt: string;
  confirmedAt: string;
  completedAt: string;
  cancelledAt: string;
  createdAt: string;
  updatedAt: string;
  professional: ProfessionalDto;
}
